### ISSUE TITLE
`diffuse` BSDF with textured reflectance not optimizing

### ISSUE THREAD

**Issue by andrew-shc** [mitsuba-renderer/mitsuba3#1699]
We are trying to perform a gradient-based optimization on the albedo (textured, via `.ply` attributes)  of a sphere; however, the loss slowly bounces back after certain amount of iterations. The loss is computed via MSE between the pixels of the rendered image and the synthetic ground truth image (also rendered via Mitsuba). We made sure the transformations are exactly the same and followed the gradient-based optimization tutorial.

Is there anything we are missing with the textured materials, optimizer settings, etc.? Thank you.

Here's our Python scene:
```python
base_scene = mi.load_dict({
        "type": "scene",
        # geometry
        "geometry": {
            "type": "ply",
            "filename": "simple_cube_data/joined_initial.ply",
            "bsdf": {
                "type": "diffuse",
                "reflectance": {
                    'type': 'mesh_attribute',
                    'name': 'vertex_color'
                },
            }
        },
        # env. lighting
        "emitter": {
            "type": "envmap",
            "bitmap": mi.Bitmap(
                envmap_data,
                mi.Bitmap.PixelFormat.RGB
            ),
        },
        "integrator": {"type": "path"}
    })
```

Training loop:
```python
N = mi.traverse(base_scene)["geometry.vertex_positions"].shape[0]//3

iteration_count = 1000
errors = []
for it in range(iteration_count):
    image = mi.render(base_scene, base_params, spp=4, sensor=sensor)

    loss = mse(image)

    dr.backward(loss)
    opt.step()
    opt[key] = dr.clip(opt[key], 0.0, 1.0)
    base_params.update(opt)

    color_generated = dr.reshape(dr.cuda.ad.TensorXf, base_params[key], (N, 3))
    param_ref_tensor = dr.cuda.ad.TensorXf([[param_ref[0][0], param_ref[1][0], param_ref[2][0]]])
    err_ref = dr.sum(dr.square(param_ref_tensor - color_generated))
    print(f"Iteration {it:02d}: parameter error = {err_ref}")#, end='\r')
    errors.append(np.array(dr.slice(err_ref)))
```

Base scene:

<img width="512" height="512" alt="Image" src="https://github.com/user-attachments/assets/185ba1ac-4df3-4a7c-bf8e-b3b6e71719dc" />

Final scene (looks mostly as intended, but you would notice colors of the sphere gets more noiser over iterations):

<img width="512" height="512" alt="Image" src="https://github.com/user-attachments/assets/652381e5-054b-4e2e-a707-760c042986eb" />

Error history:

<img width="598" height="453" alt="Image" src="https://github.com/user-attachments/assets/75b9bcb3-2db0-4897-a027-2812f4171548" />

**Comment by merlinND** [comment]
Hello @andrew-shc,

If your problem is resolved, could you please share the solution so that others who run into something similar can benefit from your experience?

**Comment by andrew-shc** [comment]
I was comparing against the ground truth parameter instead of the loss, but I did find increasing the SPP helped the optimizer find ground truth solution better.

**Comment by njroussel** [comment]
Quick note: it's strongly recommended to change your rendering seed on every iteration. Keeping the same seed can (depending on the setup) create the same Monte Carlo noise pattern at each iteration. Ultimately, this means that your scene parameters will try to compensate for that rendering noise.
